api/clients/v2/payloaddispersal/payload_disperser.go

```diff
@@ -113,35 +145,49 @@ func (pd *PayloadDisperser) SendPayload(
 
 	probe.SetStage("QUEUED")
 
+	// poll the disperser for the status of the blob until its received adequate signatures in regards to
+	// confirmation thresholds, a terminal error, or a timeout
 	timeoutCtx, cancel = context.WithTimeout(ctx, pd.config.BlobCompleteTimeout)
 	defer cancel()
-	blobStatusReply, err := pd.pollBlobStatusUntilSigned(timeoutCtx, blobKey, blobStatus.ToProfobuf(), probe)
+	blobStatusReply, err := pd.pollBlobStatusUntilSigned(timeoutCtx, blobKey, confirmationThreshold, blobStatus.ToProfobuf(), probe)
 	if err != nil {
 		return nil, fmt.Errorf("poll blob status until signed: %w", err)
 	}
 
 	pd.logSigningPercentages(blobKey, blobStatusReply)
-	probe.SetStage("build_cert")
 
-	eigenDACert, err := pd.buildEigenDACert(ctx, blobKey, blobStatusReply)
+	probe.SetStage("build_cert")
+	err = pd.blockMonitor.WaitForBlockNumber(ctx, blobStatusReply.SignedBatch.Header.ReferenceBlockNumber)
```

Comment:
Probably the block monitor isn't needed here either, since you should specify block number in `CallOpts`, whenever making an eth call which needs to happen at/after a given block

Block monitor should just be deleted, if you agree with my assessment

Reply:
true but it does some heuristic that better ensures the call to succeed with higher probability

Reply:
Hmm. I was thinking that if you specify a block number in `CallOpts`, the call would wait until the block is reached. But I see now that it will just fail.

I think the best solution might be to make a utility, which will:
- wait for a given block to arrive
- try the call
- retry the call if it fails due to block number in the future (if the node that serves to call happens to be slightly behind the node that supplied the block number)

But I don't think it makes sense to do now, as-is probably works fine, unless we see related failures

Reply:
[0d0f7bb](https://github.com/Layr-Labs/eigenda/pull/1532/commits/0d0f7bbd771bccdab89ebb767b78dcd2163673bb)